api/v1alpha1/baseboardmanagement_types.go

```diff
@@ -17,29 +17,63 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// BaseboardManagementState represents the BaseboardManagement state.
+type BaseboardManagementState string
+
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // BaseboardManagementSpec defines the desired state of BaseboardManagement
 type BaseboardManagementSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Host is the host IP address of the BaseboardManagement.
+	// +kubebuilder:validation:MinLength=1
+	Host string `json:"host"`
+
+	// AuthSecretRef is the SecretReference that contains authentication information of the BaseboardManagement.
+	// The Secret must contain username and password keys.
+	AuthSecretRef corev1.SecretReference `json:"authSecretRef"`
+
+	// Vendor is the vendor name of the BaseboardManagement.
+	// +kubebuilder:validation:MinLength=1
+	Vendor string `json:"vendor"`
+
+	// Power is the desired power state of the BaseboardManagement.
+	// +kubebuilder:validation:MinLength=1
+	Power string `json:"power"`
+
+	// Boot is the desired boot device for the BaseboardManagement.
+	Boot Boot `json:"boot"`
+}
+
+type Boot struct {
+	// BootDevice is the desired boot for the BaseboardManagement.
+	// +kubebuilder:validation:MinLength=1
+	BootDevice string `json:"bootDevice"`
```

Comment:
Actually I did not have this as a list because I saw bmc-lib can only set the first boot dev. 
I think we might need to add functionality to set the complete boot order?